### PR TITLE
Update release compatibility table

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) improves the version compatibility content:

- Added 2.3.4 Commerce
- Added several Page Builder versions
- Fixed matrix to reflect correct compatibility for Inventory and Page Builder

## Affected DevDocs pages

-  https://devdocs.magento.com/release/

## Additional info

To help future contributors make accurate updates, consider the following when making updates:

- There is a 1:1 relationship between versions of Magento and Page Builder. Later versions of Page Builder are **not supported** on earlier versions of Magento. The same applies to VBEs.
- Later versions of Inventory are **compatible** with earlier versions of Magento (and vice versa).

### CODE REVIEW

File: src/_data/mde.yml [hunk 1]
```diff
@@ -38,83 +42,122 @@ extensions:
...
           2.3.1: general availability
           2.3.2: general availability
           2.3.3: general availability
+          2.3.4: general availability
   -
     name: Magento Inventory
     versions:
       -
-        name: 1.6.1
+        name: 1.1.4
```
Comment (dobooth) [inline src/_data/mde.yml]: This is correct?

Comment (jeff-matthews) [reply src/_data/mde.yml]: Yes, according to the canonical source for all VBEs and MDEs on the internal wiki.

Comment (jeff-matthews) [reply src/_data/mde.yml]: Confirmed that 1.6.1 doesn't exist. Also confirmed that later versions of Inventory usually work fine with earlier Magento versions and are considered "compatible" but not supported. I'll need to push some additional changes.

**Comment by jeff-matthews** [comment]
Ok, i'm done pushing changes until the next review.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**